src/abstract/Multicall.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity 0.8.13;
+
+import { IMulticall } from "src/interfaces/IMulticall.sol";
+
+/// @notice Helper utility that enables calling multiple local methods in a single call.
+/// @author Modified from Uniswap (https://github.com/Uniswap/v3-periphery/blob/main/contracts/base/Multicall.sol)
+/// License-Identifier: GPL-2.0-or-later
+abstract contract Multicall is IMulticall {
+    function multicall(bytes[] calldata data) public payable returns (bytes[] memory results) {
```

Comment:
Can this be `external` instead of `public`?

Reply:
I think so. I went to reference sushi and univ3 again and cannot find any internal use of the function. It is always called externally. So it is quite curious why they didn't make it external. This will also lower gas cost right? 

Changed to `external`.